ibis/common/graph.py

```diff
@@ -245,15 +245,67 @@ def map(self, fn: Callable, filter: Optional[Finder] = None) -> dict[Node, Any]:
         A mapping of nodes to their results.
         """
         results: dict[Node, Any] = {}
-        for node in Graph.from_bfs(self, filter=filter).toposort():
+
+        graph, _ = Graph.from_bfs(self, filter=filter).toposort()
+        for node in graph:
             # minor optimization to directly recurse into the children
             kwargs = {
                 k: _recursive_lookup(v, results)
                 for k, v in zip(node.__argnames__, node.__args__)
             }
             results[node] = fn(node, results, **kwargs)
+
         return results
 
+    @experimental
+    def map_clear(
```

Comment:
How about `map_prune` instead? `map_clear` makes me think everything gets deleted, but only no-longer-used results are cleared.

Reply:
Well, for me map_prune sounds like what you are describing. Can we merge it and maybe rename later, would like to try it out in the pandas PR :)